matscipy/fracture_mechanics/crack.py

```diff
@@ -907,13 +1276,13 @@ def set_atoms(self, atoms):
         self.update_atoms()  # now we have same u_cle in atoms and self.atoms
         min_len = min(N1_in, self.N1)
         # FIXME this assumes stable sort order for atoms and self.atoms
-        u = atoms.positions[:min_len] - self.atoms.positions[:min_len]
+        u = atoms.positions - self.atoms.positions
```

Comment:
why is this change needed?

Reply:
This was a necessary change for the case of having an extended region I which encompassed the surfaces. The original code only worked when region I was made up of the first min_len atoms, the new code works for a region I of any shape.

Reply:
ok